packages/docs/src/styles/components/navigation.scss

```diff
@@ -25,6 +25,7 @@
 
   .c-navigation__link-list {
     border-block: 1px solid var(--vertical-nav-item__border-color);
+    gap: 0;
```

Comment:
Fixing the nav panel might need its own pull request. Even with this in place, I'm seeing way more spacing between items and a narrower panel overall.

<img width="219" alt="Screenshot of the navigation panel being much narrower than before and having more vertical spacing between nav links" src="https://user-images.githubusercontent.com/7595652/222791759-b2a772bc-71c9-4a12-8bbe-2984eb3cf388.png">